### PR TITLE
(WIP) Adding support for algebraic variables in nonlinear constraints (BGH for now - not BGHP)

### PR DESCRIPTION
**Pull request by zanellia**
This PR adds support for algebraic variables in nonlinear constraints ($l_h \leq h(x,u,z(x,u)) \leq u_h$).

### CODE REVIEW

**Comment by zanellia** [comment]
@giaf I have implemented the update the update of DCt and model->fun due to algebraic variables in the BGH constraints. I get the same result with simple_dae_example.c with two different formulations (turning bounds on x into nonlinear constraints on z). I still need to check whether the residuals are computed correctly and to fix the MATLAB interface.

File: acados/ocp_nlp/ocp_nlp_constraints_bghp.c [hunk 1]
```diff
@@ -1104,6 +1111,8 @@ void ocp_nlp_constraints_bghp_config_initialize_default(void *config_)
     config->memory_set_lam_ptr = &ocp_nlp_constraints_bghp_memory_set_lam_ptr;
     config->memory_set_DCt_ptr = &ocp_nlp_constraints_bghp_memory_set_DCt_ptr;
     config->memory_set_RSQrq_ptr = &ocp_nlp_constraints_bghp_memory_set_RSQrq_ptr;
+    config->memory_set_z_alg_ptr = &ocp_nlp_constraints_bgh_memory_set_z_alg_ptr;
```
Comment (giaf) [inline acados/ocp_nlp/ocp_nlp_constraints_bghp.c]: `ocp_nlp_constraints_bghP_memory_set_z_alg_pt` plus next line plus adding the functions (which either are implemented properly or print-and-exit)

File: acados/ocp_nlp/ocp_nlp_sqp_rti.c [hunk 2]
```diff
@@ -1260,6 +1259,7 @@ int ocp_nlp_sqp_rti(void *config_, void *dims_, void *nlp_in_, void *nlp_out_,
     for (ii = 0; ii <= N; ii++)
     {
         config->constraints[ii]->memory_set_ux_ptr(nlp_out->ux+ii, mem->constraints[ii]);
+		config->constraints[ii]->memory_set_dzdux_tran_ptr(mem->dzduxt+ii, mem->constraints[ii]);
```
Comment (giaf) [inline acados/ocp_nlp/ocp_nlp_sqp_rti.c]: `memory_set_z_alg_ptr` is missing I guess

File: acados/ocp_nlp/ocp_nlp_constraints_bghp.c (hunk 1 above)
Comment (zanellia) [reply acados/ocp_nlp/ocp_nlp_constraints_bghp.c]: addressed here 3a47d76a0 and here 3a47d76a0.

File: acados/ocp_nlp/ocp_nlp_constraints_bghp.h [hunk 3]
```diff
@@ -147,10 +148,15 @@ struct blasfeo_dvec *ocp_nlp_constraints_bghp_memory_get_fun_ptr(void *memory_);
 struct blasfeo_dvec *ocp_nlp_constraints_bghp_memory_get_adj_ptr(void *memory_);
 //
 void ocp_nlp_constraints_bghp_memory_set_ux_ptr(struct blasfeo_dvec *ux, void *memory_);
+//
 void ocp_nlp_constraints_bghp_memory_set_lam_ptr(struct blasfeo_dvec *lam, void *memory_);
 //
 void ocp_nlp_constraints_bghp_memory_set_DCt_ptr(struct blasfeo_dmat *DCt, void *memory);
 //
+void ocp_nlp_constraints_bgh_memory_set_z_alg_ptr(struct blasfeo_dvec *z_alg, void *memory_);
```
[the remark this answers is not in the document]
Comment (zanellia) [reply acados/ocp_nlp/ocp_nlp_constraints_bghp.h]: addressed here 3a47d76a0 and here 3a47d76a0.

File: acados/ocp_nlp/ocp_nlp_sqp_rti.c (hunk 2 above)
Comment (zanellia) [reply acados/ocp_nlp/ocp_nlp_sqp_rti.c]: addressed here 55fb99bc14

**Comment by giaf** [comment]
ok good, merging

**Pull request merged**